tests/ensemble_evaluator/test_prefect_storage.py

```diff
@@ -20,29 +20,35 @@ def test_storage_driver(tmpdir):
 
         # Check storage path
         global_storage_path = storage.get_storage_path(None)
-        expected_global_storage_path = os.path.join(storage_path, "global")
+        expected_global_storage_path = storage_path / "global"
         assert expected_global_storage_path == global_storage_path
 
         storage_path_iens_1 = storage.get_storage_path(1)
-        expected_storage_path = os.path.join(storage_path, "1")
+        expected_storage_path = storage_path / "1"
         assert expected_storage_path == storage_path_iens_1
 
         # Store global file
         global_resource_uri = storage.store(file_name)
-        expected_uri = os.path.join(storage_path, "global", file_name)
+        expected_uri = storage_path / "global" / file_name
         assert expected_uri == global_resource_uri
-        assert os.path.isfile(global_resource_uri)
+        assert global_resource_uri.is_file()
 
         # Store file for realization 1
         real_resource = storage.store(file_name, 1)
-        expected_uri = os.path.join(storage_path, "1", file_name)
+        expected_uri = storage_path / "1" / file_name
         assert expected_uri == real_resource
-        assert os.path.isfile(real_resource)
+        assert real_resource.is_file()
 
         # Retrieve resource from global storage
         storage.retrieve(global_resource_uri, target="global_resource")
-        assert os.path.isfile("global_resource")
+        assert Path("global_resource").is_file()
 
         # Retrieve resource from realization storage
         storage.retrieve(real_resource, target="realization_resource")
-        assert os.path.isfile("realization_resource")
+        assert Path("realization_resource").is_file()
+
+        # Retrieve non-existing file
+        with pytest.raises(
+            ValueError, match=f"Storage driver can't handle file: {Path('foo')}"
+        ):
+            storage.retrieve("foo")
```

Comment:
This fails, not because the file does not exist, but because the path is not relative to the `storage_path`. How about doing something like:

```python
    file_uri = global_storage_path / "foo"
    assert not file_uri.is_file()
    with pytest.raises(
            ValueError, match=f"Storage driver can't handle file: {file_uri}"
        ):
            storage.retrieve(file_uri)
```

instead? :)

Reply:
Yes that relates to my comment up, about the error handling in storage, that will be fixed.